### DISCUSSION TITLE
Restart after 7 s

### DISCUSSION THREAD

**Discussion by Jerby13** [ArminJo/Ultimate-Battery-Tester#1] [General]
Hi Armin,

Great project, so I started to build the tester myself. But unfortunately I'm having problems to get it working properly. From first build I'm getting strange behaviour and ended up with I think wrong data in the EEPROM which now results in restarts after the first init and no matter what I write into the EEPROM it keeps restarting.

Some questions:
Could you add a software definition to enable/disable the reading/writing to EEPROM?
What versions of the libraries that are not included in Github do you use?
What data should be written to EEPROM in initial state, 0xFF?
I tested the software with nano and promini 5v/16Mhz.

I hope that you can get me on the way,
Regards,
Jerby

**Comment by Jerby13** [comment]
I get the following monitor output:

START D:\Downloads\Ultimate-Battery-Tester-master\UltimateBatteryTester\UltimateBatteryTester.ino
Version 3.2 from Nov 20 2023
Button pin=2
Connect pin 10 to ground, to suppress such prints not suited for Arduino plotter
674 compressed EEPROM ����:6
0 0 0
EEPROM values=0x0 0x0 0x0
65506 65506 6550���4566 65476
EEPROM values=0x0 0x0 0x0
65446 65446 654���56516 65416
EEPROM values=0x0 0x0 0x0
65386 65386 65���67�356 65356
EEPROM values=0x0 0x0 0x0
65326 65326 6���7865296 65296

....

45494 45436 45436
EEPROM values=0x0 0x0 0x0
���8945406
45434 45376 45376
EEPROM values=0x0 0x0 0x0���67645346
Voltage=0.000V->45.374V:45374 Current=0mA->���K��ESR=0.000ohm->45.316ohm:45316 LoadResistor=0.000ohm���4c84mAh Duration=11h_14min
Stored minu�!��K�START D:\Downloads\Ultimate-Battery-Tester-master\UltimateBatteryTester\UltimateBatteryTester.ino
Version 3.2 from Nov 20 2023
Button pin=2
Connect pin 10 to ground, to suppress such prints not suited for Arduino plotter
674 compressed EEPRO���567d
0 0 0
EEPROM values=0x0 0x0 0x0
65506 65506 65���67�476 65476
EEPROM values=0x0 0x0 0x0
65446 65446 6�4�7865416 65416

**Reply by ArminJo** [reply]
Non initialized EEPROM is always 0xFF.
Is your supply voltage OK because the Serial output of Serial.println(F("compressed EEPROM values found")); is partially missing.
You can try to comment out the line 443 `readAndProcessEEPROMData(false);`

**Comment by Jerby13** [comment]
I changed the I2C LCD with a simple parallel version and the restarts are over!

**Reply by ArminJo** [reply]
Thanks for reporting, my home version is a parallel LCD.
I will check it with a I2C version.

**Comment by Jerby13** [comment]
I also switched to Arduino IDE today and before I used Platform IO and had instability issues with parallel and I2C.

**Comment by Jerby13** [comment]
Found the root cause of the instability issues, the button interrupt pin 2. If a wire is connected to pin 2 all kind of issues occur. As far as I can see internal pull-up is enabled at pin 2 or not? I'll test it with external pull-up.

**Reply by ArminJo** [reply]
Thanks,
You must disable the line #define BUTTON_IS_ACTIVE_HIGH if you use a button connecting to ground.

**Reply by Jerby13** [reply]
Oeps, incredible that I didn't see that clear comment in your code. It explains all the weird responses.
Please make a note on the schematic that the default setting in the code is active high. 
Thanks for your very quick respons, it now is working great.

**Reply by ArminJo** [reply]
I changed the default settings to low, because of your feedback 😀 .
